kotlin/src/main/kotlin/org/jdbi/v3/core/kotlin/KotlinMapper.kt

```diff
@@ -11,108 +11,190 @@
+            val memberPropertiesWithValues = memberPropertyMappers.mapValues { (_, mapper) ->
+                mapper.map(r, c)
+            }
 
-                propertiesWithValue.forEach { (prop, value) ->
+            constructor.isAccessible = true
+            constructor.callBy(constructorParametersWithValues).also { instance ->
+                memberPropertiesWithValues.forEach { (prop, value) ->
                     prop.isAccessible = true
                     prop.setter.call(instance, value)
                 }
-
-                instance
-            } catch (e: InvocationTargetException) {
-                throw IllegalArgumentException("A bean, ${kClass.simpleName} was mapped which was not instantiable", e.targetException)
-            } catch (e: ReflectiveOperationException) {
-                throw IllegalArgumentException("A bean, ${kClass.simpleName} was mapped which was not instantiable", e)
             }
         }
+    }
 
+    private fun getConstructorParameterProvider(rs: ResultSet,
+                                                ctx: StatementContext,
+                                                parameter: KParameter,
+                                                columnNames: List<String>,
+                                                columnNameMatchers: List<ColumnNameMatcher>,
+                                                unmatchedColumns: MutableSet<String>
+    ): RowMapper<*> {
+        val parameterName = parameter.paramName()
+
+        val nested = parameter.findAnnotation<Nested>()
+
+        return if (nested == null) {
+            val columnIndex = findColumnIndex(parameterName, columnNames, columnNameMatchers, { parameter.name })
+                    .orElseThrow({
+                        IllegalArgumentException(
+                                "Constructor '${constructor.name}' parameter '$parameterName' has no column in the result set. " +
+                                        "Verify that the Java compiler is configured to emit parameter names, " +
+                                        "that your result set has the columns expected, or annotate the " +
+                                        "parameter names explicitly with @ColumnName"
+                        )
+                    })
+
+            val type = parameter.type.javaType
+
+            ctx.findColumnMapperFor(type)
+                    .map { mapper -> SingleColumnMapper(mapper, columnIndex + 1) }
+                    .orElseThrow {
+                        IllegalArgumentException(
+                                "Could not find column mapper for type '$type' of parameter " +
+                                        "'$parameter' for constructor '$constructor'")
+                    }
+                    .also {
+                        unmatchedColumns.remove(columnNames[columnIndex])
+                    }
+        } else {
+            //TODO: Support for nested non-kotlin types?
```

Comment:
I wouldn't bother. The reflective mappers in core only support nested mappers of the same type.

Reply:
That's what I figured, I'll remove the 'TODO's.